datadog/resource_datadog_monitor.go

```diff
@@ -632,45 +599,88 @@ func resourceDatadogMonitorRead(d *schema.ResourceData, meta interface{}) error
 	sort.Strings(tags)
 
 	log.Printf("[DEBUG] monitor: %+v", m)
-	d.Set("name", m.GetName())
-	d.Set("message", m.GetMessage())
-	d.Set("query", m.GetQuery())
-	d.Set("type", m.GetType())
-	d.Set("priority", m.GetPriority())
-	d.Set("restricted_roles", m.GetRestrictedRoles())
+	if err := d.Set("name", m.GetName()); err != nil {
```

Comment:
Does this error if the value being set is empty, or is it just if an internal TF SDK piece fails?

Reply:
It errors if an internal SDK piece fails, like wrong type or something